### DISCUSSION TITLE
Error when trying to set password for a user

### DISCUSSION THREAD

**Discussion by mmccorkindale** [broadinstitute/seqr#3957] [Q&A] [answered]
Hi,

I'm setting up a local instance of seqr, and so far, we have successfully managed to create a superuser, login, and perform a search on some test data. We can also create new users with email addresses, but we're having problems when users try to either login, or set passwords.

When attempting to login as anyone other than the superuser (with or without the passwords mentioned below):
" . Invalid credentials"

When attempting to send a reset link using 'Forgot Password':
" . Postmark API response 401 (Unauthorized): { "ErrorCode": 10, "Message": "Request does not contain a valid Server token."}

I've looked at the postgres auth_user table, and I can see all of the users created so far, along with their email addresses and (pre-assigned?) passwords. The only difference I can see is that the superuser has an sha256 hash in the 'password' column, whereas the other created accounts have a string of characters preceded by '!'. We've tried logging in using this string of characters as the password, but is this some kind of set password token? In our previous local instance I believe we set passwords by sending the user a link such as:

http:/ /server-name:port/set-password?token=token_characters

The link failed to send in the old instance, but we were able to find it in the gunicorn.log file. Is there a similar/equivalent link in the latest seqr version, or is the issue something different?

Thanks,

Michael

**Comment by hanars** [comment]
Hi Micheal,

That error is being thrown because you have not configured an email server for seqr but are trying to send a "forgot password" email. If you have your own postmark account set up, you can add a `POSTMARK_SERVER_TOKEN` environment variable to your configuration and this will allow seqr to automatically send emails at whatever billing rate you have set up for that account. However, most people who set seqr up on their own simply do not set up email as an option, and just opt not to use any of the functionality that requires email

The link you describe above is close to the link users should be using to set their password, but it is not quite right. The URL should be `http://server-name:port/login/set-password/<token_characters>` where token_characters will just be that "string of characters preceded by '!'" you describe above

Let me know if you have any further questions!

**Reply by mmccorkindale** [reply]
Hi Hana,

We're getting a 404 when attempting to use the set-password format you provided along with our token - we looked at the traceback and it seems to be set_password instead, but using this gives us a different error:

"Not Found The requested resource was not found on this server."

Including the '!' preceding the token causes a blank page to be displayed. Do the tokens expire after a certain amount of time?

Thanks,

Michael

**Reply by hanars** [reply]
the blank page is probably the correct page but with some sort of javascript error. Can you check the browser console and see what it says?

**Reply by mmccorkindale** [reply]
Hi Hana,

See below for the error we're getting in the console:

```
initialJSON:  Object                          <password_token>
Creating store with initial state: Object                      app-cedf3e53.js:1 
TypeError: o is not a function                                    app-cedf3e53.js:1 
    at app-cedf3e53.js:1:116481
    at Array.forEach (<anonymous>)
    at L.A.concat.a.reduce.e.concat.j.<computed> (app-cedf3e53.js:1:116454)
    at app-cedf3e53.js:1:116746
    at Array.reduce (<anonymous>)
    at L (app-cedf3e53.js:1:116364)
    at Object.resumeValidation (app-cedf3e53.js:1:123655)
    at app-cedf3e53.js:1:382045
    at Fu (app-cedf3e53.js:1:360683)
    at t.unstable_runWithPriority (app-cedf3e53.js:1:378630)
```

There is a second error with the same traceback, but it says:

```
Uncaught TypeError: o is not a function
```

On the third line above instead.

Thanks,

Michael

**Reply by hanars** [reply, marked as answer]
ah yes, this does appear to be a seqr bug. In the meantime while this gets fixed, you can reset the password for people using the admin interface: `http://server-name:port/admin/auth/user/` 
Select the user to update, and then where it shows the password there should be a link to a form to change the password

**Comment by AunErki** [comment]
> I've looked at the postgres auth_user table, and I can see all of the users created so far, along with their email addresses and (pre-assigned?) passwords.

Hi. Could you please describe how did you got to look into postgres auth_user table? Unfortunately,  I am unable to find this table.

**Reply by hanars** [reply]
In general it is bad practice to interact directly with the postgres database, so you should use the `http://server-name:port/admin/auth/user/` to view an interact with the user data

**Comment by AunErki** [comment]
Hello Hana and _mmccorkindale_. I am experiencing exactly the same problem after recent docker install. The set password and reset password links lead to an empty white page. However, I discovered that actually there is correct page and form for a fraction of second, but it disappears momentarily. So, basically I can see it, when I refresh page really fast. Is there  anything I could do or try out to fix this?

**Reply by hanars** [reply]
So unless you are willing to contribute code to the seqr repository, then no. As I stated above, this is a bug in seqr so there is no way to fix it without actually fixing the code. In the meantime, as I stated above, you can reset passwords from the admin view

**Comment by hanars** [comment]
Just wanted to update that the javascript bug causing this issue has been fixed in the latest seqr release
